examples/app/pivot/main.py

```diff
@@ -0,0 +1,377 @@
+    wdg['y_major_label_size'] = bmw.TextInput(title='Y Labels Font Size', value=str(PLOT_AXIS_LABEL_SIZE), css_classes=['wdgkey-y_major_label_size', 'adjust-drop'])
+    wdg['circle_size'] = bmw.TextInput(title='Circle Size (Dot Only)', value=str(CIRCLE_SIZE), css_classes=['wdgkey-circle_size', 'adjust-drop'])
+    wdg['bar_width'] = bmw.TextInput(title='Bar Width (Bar Only)', value=str(BAR_WIDTH), css_classes=['wdgkey-bar_width', 'adjust-drop'])
+    wdg['line_width'] = bmw.TextInput(title='Line Width (Line Only)', value=str(LINE_WIDTH), css_classes=['wdgkey-line_width', 'adjust-drop'])
+    wdg['download'] = bmw.Button(label='Download csv', button_type='success')
+    wdg['export_config'] = bmw.Div(text='Export Config to URL', css_classes=['export-config'])
+
+    wdg['series_legend'].text = build_series_legend()
+
+    #use wdg_config (from 'widgets' parameter in URL query string) to configure widgets.
+    if init['init_load']:
+        for key in wdg_config:
+            if key in wdg:
+                if hasattr(wdg[key], 'value'):
+                    wdg[key].value = str(wdg_config[key])
+                elif hasattr(wdg[key], 'active'):
+                    wdg[key].active = wdg_config[key]
+        init['init_load'] = False
+
+    wdg['data'].on_change('value', update_data)
+    wdg['chart_type'].on_change('value', update_sel)
+    wdg['x'].on_change('value', update_sel)
+    wdg['x_group'].on_change('value', update_sel)
+    wdg['y'].on_change('value', update_sel)
+    wdg['y_agg'].on_change('value', update_sel)
+    wdg['series'].on_change('value', update_sel)
+    wdg['series_stack'].on_change('value', update_sel)
+    wdg['explode'].on_change('value', update_sel)
+    wdg['explode_group'].on_change('value', update_sel)
+    wdg['plot_title'].on_change('value', update_sel)
+    wdg['plot_title_size'].on_change('value', update_sel)
+    wdg['plot_width'].on_change('value', update_sel)
+    wdg['plot_height'].on_change('value', update_sel)
+    wdg['opacity'].on_change('value', update_sel)
+    wdg['x_min'].on_change('value', update_sel)
+    wdg['x_max'].on_change('value', update_sel)
+    wdg['x_scale'].on_change('value', update_sel)
+    wdg['x_title'].on_change('value', update_sel)
+    wdg['x_title_size'].on_change('value', update_sel)
+    wdg['x_major_label_size'].on_change('value', update_sel)
+    wdg['x_major_label_orientation'].on_change('value', update_sel)
+    wdg['y_min'].on_change('value', update_sel)
+    wdg['y_max'].on_change('value', update_sel)
+    wdg['y_scale'].on_change('value', update_sel)
+    wdg['y_title'].on_change('value', update_sel)
+    wdg['y_title_size'].on_change('value', update_sel)
+    wdg['y_major_label_size'].on_change('value', update_sel)
+    wdg['circle_size'].on_change('value', update_sel)
+    wdg['bar_width'].on_change('value', update_sel)
+    wdg['line_width'].on_change('value', update_sel)
```

Comment:
since these are all getting the same callback, it might be a little more economical in terms of code to use a loop, somethign like:
```
for name in widget_names:
    wdg[name].on_change('value', update_sel)
```